becquerel/core/rebin.py

```diff
@@ -394,63 +301,65 @@ def rebin(in_spectra, in_edges, out_edges, method="interpolation",
         The rebinned spectrum/a
     """
     method = method.lower()
+    if not hasattr(in_spectra, "shape"):
+        assert xarray is False, "Use xarray as input not list"
+        in_spectra = np.asarray(in_spectra)
     # Cast data types and check listmode input
     if method == "listmode":
-        if not np.all(in_spectra >= 0.):
+        if (in_spectra < 0).any():
             raise RebinError('Cannot rebin spectra with negative values with '
                              'listmode method')
-        if np.all(in_spectra < 1):
+        if (in_spectra < 1).all():
             raise RebinError('Cannot rebin spectra with all values less than '
                              'one with listmode method')
         if np.issubdtype(in_spectra.dtype.type, np.floating):
-            if np.allclose(in_spectra, np.round(in_spectra)):
-                # Don't warn in the case of floats which round to integers
-                pass
-            else:
-                warnings.warn(
-                    'Argument in_spectra contains float value(s) which ' +
-                    'will have decimal precision loss when converting to ' +
-                    'integers for rebin method listmode.',
-                    RebinWarning)
-            in_spectra = np.asarray(np.round(in_spectra), dtype=np.int64)
-    else:
-        in_spectra = np.asarray(in_spectra, dtype=np.float64)
-    in_edges = np.asarray(in_edges, np.float64)
-    out_edges = np.asarray(out_edges, np.float64)
-    if slopes is None:
-        slopes = np.zeros_like(in_spectra, dtype=np.float64)
-    else:
-        slopes = np.asarray(slopes, dtype=np.float64)
-    # Broadcast 1D -> 2D if necessary
-    in_edges = _broadcast(in_edges, in_spectra.shape)
-    slopes = _broadcast(slopes, in_spectra.shape)
-    # Check dimensions
-    _check_ndim(in_spectra, {1, 2}, 'in_spectra')
-    _check_ndim(in_edges, {1, 2}, 'in_edges')
-    _check_ndim(slopes, {1, 2}, 'slopes')
-    _check_ndim(out_edges, 1, 'out_edges')
-    # Check shape
-    _check_shape(in_spectra, in_edges, 'in_spectra', 'in_edges', edges=True)
-    _check_shape(in_spectra, slopes, 'in_spectra', 'slopes')
+            # np.rint is a ufunc: natively supported by xarray:
+            in_spectra_rint = np.rint(in_spectra)
+            # TEMP disable this test, always raise warning for now because
```

Comment:
We can remove this now right?

Reply:
Not sure what part of the code you refer too. I think I would leave the check for `int` type and warn if not. It would be great if we could leave the `isclose` part as well, but as @chunhochow mention in the comment that would break compatibility with data types that only accept `ufuncs`. Thoughts?

Reply:
I'm referring to the `TEMP` tag which can be resolved before this is merged. I this the `int` type warning is 👍

Reply:
Expanding on that: lets either include the below code or remove it altogether. I support a try except if the allclose method doesn't work and a warning that the check cannot be performed. What edge cases would this note work for?